src/Client/WorkflowClient.php

```diff
@@ -276,17 +276,28 @@ public function listWorkflowExecutions(
                 $request->setNextPageToken($nextPageToken);
             } while ($nextPageToken !== '');
         };
-        $counter = function () use ($namespace, $query): int {
-            $response = $this->client->CountWorkflowExecutions((new CountWorkflowExecutionsRequest())
-                ->setNamespace($namespace)
-                ->setQuery($query));
-
-            return (int)$response->getCount();
-        };
+        $counter = fn(): int => $this->countWorkflowExecutions($query, $namespace);
 
         return Paginator::createFromGenerator($loader($request), $counter);
     }
 
+    /**
+     * @inheritDoc
+     */
+    public function countWorkflowExecutions(
+        string $query,
+        string $namespace = 'default',
+    ): int {
```

Comment:
This should not return an int as we have new result info coming at https://github.com/temporalio/api/pull/303 that can returned grouped counts

Reply:
Got it. Will be a DTO